float64 opp_gap_direction